js/tuxexchange.js

```diff
@@ -0,0 +1,567 @@
+            'symbol': symbol,
+            'type': type,
+            'side': side,
+            'price': price,
+            'amount': amount,
+            'cost': undefined,
+            'average': undefined,
+            'filled': undefined,
+            'remaining': undefined,
+            'status': undefined,
+            'fee': undefined,
+            'trades': undefined,
+        };
+    }
+
+    async cancelOrder (id, symbol = undefined, params = {}) {
+        if (symbol === undefined) {
+            throw new ArgumentsRequired (this.id + ' cancelOrder () requires a "symbol" argument');
+        }
+        await this.loadMarkets ();
+        const ids = this.getIdsFromSymbol (symbol);
+        const cancelRequest = {
+            'method': 'cancelorder',
+            'id': id,
+            'market': this.currencyId (ids.base),
+        };
+        const result = await this.privatePost (cancelRequest);
+        return { 'info': result };
+    }
+
+    parseOrder (order) {
+        const symbol = this.findSymbol (this.safeString (order, 'market_pair'));
+        const orderDate = this.parseDate (this.safeString (order, 'date'));
+        const timestamp = orderDate.valueOf ();
+        const amount = this.asFloat (this.safeString (order, 'amount'));
+        const filled_amount = this.asFloat (this.safeString (order, 'filledamount'));
+        return {
+            'info': order,
+            'id': this.safeString (order, 'id'),
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'symbol': symbol,
+            'type': 'limit',
+            'side': this.safeString (order, 'type'),
+            'price': this.asFloat (this.safeString (order, 'price')),
+            'amount': amount,
+            'remaining': amount - filled_amount,
+            'filled': filled_amount,
+            'status': 'open',
+            'lastTradeTimestamp': undefined,
+        };
+    }
+
+    async fetchOpenOrders (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const openOrdersResp = await this.privatePost ({ 'method': 'getmyopenorders' });
+        const openOrderMap = this.parseJson (openOrdersResp);
+        const openOrders = this.values (openOrderMap);
+        let market = undefined;
+        if (symbol !== undefined) {
+            market = this.getMarket (symbol);
+        }
+        const result = this.parseOrders (openOrders, market, since, limit);
+        return result;
+    }
+
+    async fetchMyTrades (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        const tradeHistoryResp = await this.privatePost ({ 'method': 'getmytradehistory' });
+        const myTrades = this.parseJson (tradeHistoryResp);
+        let results = this.parseTrades (myTrades, symbol, since, limit);
+        return results;
+    }
+
+    async fetchDepositAddress (code, params = {}) {
+        if (code === undefined) {
```

Comment:
can call `currency (code)` after `loadMarkets` to do this check

Reply:
I feel like my error message is clearer, but if you think it's more cannonical I can change it

Reply:
whatever you prefer, I was thinking it might be more canonical though